src/web3-adapter/eth_gasStation.ts

```diff
@@ -0,0 +1,29 @@
+import https from "https";
+
+export function patchEthGasStation(web3: any): void {
+  web3.ethGasSation = (apiKey: string) => {
+    return new Promise((resolve, reject) => {
+      https.get(
+        {
+          hostname: "data-api.defipulse.com",
+          path: "/api/v1/egs/api/ethgasAPI.json",
+          search: `?api-key=${apiKey}`,
+        },
+        (response) => {
+          let data = "";
+          response.on("data", (chunk) => {
+            data += chunk;
```

Comment:
Repeatedly adding to a string is O(n^2). While the runtime might be able to optimize this away in a loop, its unlikely to be able to do so in a callback which triggers over multiple executions of the event loop like this one.

Instead, append each chunk to a list of strings `chunks`, then call `resolve(data.join(""))` at the end.

Reply:
This is the type of quality comment content I have come to expect from you.